src/Nethermind/Nethermind.Blockchain/Spec/ChainHeadSpecProvider.cs

```diff
@@ -42,6 +45,26 @@ public ChainHeadSpecProvider(ISpecProvider specProvider, IBlockFinder blockFinde
 
         public long[] TransitionBlocks => _specProvider.TransitionBlocks;
         
-        public IReleaseSpec GetSpec() => GetSpec(_blockFinder.FindBestSuggestedHeader()?.Number ?? 0);
+        public IReleaseSpec GetSpec()
+        {
+            long headerNumber = _blockFinder.FindBestSuggestedHeader()?.Number ?? 0;
+            if (headerNumber == _lastHeader)
+            {
+                IReleaseSpec releaseSpec = _headerSpec;
+                if (headerNumber == _lastHeader)
+                {
+                    if (releaseSpec is not null)
+                    {
+                        return releaseSpec;
+                    }
+                }
+            }
+
+            lock (_lock)
```

Comment:
Again, what if headerNumber changed between ln50 and ln65?
We will overwrite _headerSpec with the previous one.